### PR TITLE
Make sure that mach gets the correct paths of executables in virtualenv

### LINKED ISSUE #13141: OSError: [Errno 13] Permission denied: '/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/mach'
After a few months away, I just pulled and now, I get the following when attempting to build

```sh
$ ./mach build
Pip failed to execute properly: Exception:
Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pip/commands/install.py", line 317, in run
    prefix=options.prefix_path,
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pip/req/req_set.py", line 742, in install
    **kwargs
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pip/req/req_install.py", line 831, in install
    self.move_wheel_files(self.source_dir, root=root, prefix=prefix)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pip/req/req_install.py", line 1032, in move_wheel_files
    isolated=self.isolated,
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pip/wheel.py", line 346, in move_wheel_files
    clobber(source, lib_dir, True)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pip/wheel.py", line 317, in clobber
    ensure_dir(destdir)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pip/utils/__init__.py", line 83, in ensure_dir
    os.makedirs(path)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/os.py", line 157, in makedirs
    mkdir(name, mode)
OSError: [Errno 13] Permission denied: '/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/mach'
```

For some reason, `pip` seems to want to install (or use) mach in my `/opt/local`, which looks... weird.

I'm using MacOS X 10.11.6 + MacPorts.

### PR DESCRIPTION
**Pull request by ghost**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #13141 (github issue number if applicable).

- [x] These changes do not require tests because it's a cleanup related to `mach`

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/13151)

### CODE REVIEW

File: python/mach_bootstrap.py [hunk 1]
```diff
@@ -87,45 +91,35 @@ def _get_exec(names, is_valid_path=lambda _path: True):
...
-PIP_NAMES = ["pip-2.7", "pip2.7", "pip2", "pip"]
-
-
 def _activate_virtualenv(topdir):
     virtualenv_path = os.path.join(topdir, "python", "_virtualenv")
     check_exec_path = lambda path: path.startswith(virtualenv_path)
-    python = _get_exec(PYTHON_NAMES)
-    if python is None:
-        sys.exit("Python is not installed. Please install it prior to running mach.")
+    python = _get_exec_path(PYTHON_NAMES)   # If there was no python, mach wouldn't have run at all!
```
Comment (Ms2ger) [inline python/mach_bootstrap.py]: But that doesn't mean we'll find it here; keep the check, but feel free to change the error message

**Comment by Ms2ger** [comment]
r+ with the one comment fixed

File: python/mach_bootstrap.py (hunk 1 above)
Comment (Wafflespeanut) [inline python/mach_bootstrap.py]: Fixed the error message :)

**Pull request merged**